File: solutions/c/leap/1/leap.c
#include "leap.h"

bool leap_year(int n) {
    if (n % 400 == 0) {
        return true;
    } else if (n % 100 == 0) {
        return false;
    } else if (n % 4 == 0) {
        return true;
    } else {
        return false;
    }
}

File: solutions/clojure/leap/1/src/leap.clj
(ns leap)

(defn leap-year?
  "Returns true if the given year is a leap year;
  otherwise, it returns false."
  [year]
  (let [div-by? (fn [a b] (zero? (mod a b)))]
    (and (div-by? year 4)
         (or (not (div-by? year 100))
             (div-by? year 400)))))
